src/Slices/ResourceDiscovery/UI/Components/ManagedResourceLink.tsx

```diff
@@ -17,14 +18,28 @@ interface Props {
  */
 export const ManagedResourceLink: React.FC<Props> = ({ resourceUri }) => {
   if (!resourceUri) {
-    return <>-</>;
+    return <></>;
   }
 
-  const rid = resourceUri.split("/").pop();
+  const rid = extractResourceLink(resourceUri);
 
   if (!rid) {
-    return <>-</>;
+    return <></>;
   }
 
-  return <ResourceLink resourceId={rid} />;
+  return (
+    <ResourceLink
+      resourceId={rid}
+      linkText={words("discovered_resources.show_resource")}
+    />
+  );
+};
+
+// example url : /api/v2/resource/cloudflare::dns_record::CnameRecord[https://api.cloudflare.com/client/v4/,name=artifacts.ssh.inmanta.com]
+// We only want to keep the adress to the resource, which is everything that comes after /api/v2/resource/
+const extractResourceLink = (uri: string): string | null => {
```

Comment:
It's by all means correct approach but wouldn't `resourceUri.split("resource/").pop();` be enough? I feel like it's unnecessary, but if you like it better then fine for me

Reply:
We don't know if the name of the resource contains the resource/ part. the start of the uri is a fixed element we can go by.